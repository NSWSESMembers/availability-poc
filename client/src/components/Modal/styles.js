import { StyleSheet } from 'react-native';

const PADDING = 8;
const BORDER_RADIUS = 5;
const FONT_SIZE = 16;
const HIGHLIGHT_COLOR = 'rgba(0,118,255,0.9)';

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    alignItems: 'center',
    justifyContent: 'center',
  },
  overlayStyle: {
    flex: 1,
    padding: '5%',
    justifyContent: 'center',
    backgroundColor: 'rgba(0,0,0,0.7)',
  },

  optionContainer: {
    borderRadius: BORDER_RADIUS,
    flexShrink: 1,
    marginTop: 8,
    marginBottom: 8,
    padding: PADDING,
    backgroundColor: 'rgba(255,255,255,0.8)',
  },

  cancelContainer: {
    alignSelf: 'stretch',
  },

  nextContainer: {
    alignSelf: 'stretch',
  },


  headerContainer: {
    alignSelf: 'stretch',
  },

  selectStyle: {
    borderColor: '#ccc',
    borderWidth: 1,
    padding: PADDING,
    borderRadius: BORDER_RADIUS,
  },

  selectTextStyle: {
    textAlign: 'center',
    color: '#333',
    fontSize: FONT_SIZE,
  },

  headerStyle: {
    borderRadius: BORDER_RADIUS,
    backgroundColor: 'rgba(255,255,255,0.8)',
    padding: PADDING,
  },

  headerTextStyle: {
    textAlign: 'center',
    color: '#333',
    fontSize: FONT_SIZE,
  },

  headerSubTextStyle: {
    textAlign: 'center',
    color: '#333',
    fontSize: FONT_SIZE - 4,
  },

  etaInput: {
    height: 100,
    fontSize: 20,
    color: 'red',
    textAlign: 'center',
  },

  nextStyle: {
    borderRadius: BORDER_RADIUS,
    backgroundColor: 'rgba(255,255,255,0.8)',
    padding: PADDING,
    marginBottom: 8,
  },

  nextTextStyle: {
    textAlign: 'center',
    color: 'green',
    fontSize: FONT_SIZE,
  },

  cancelStyle: {
    borderRadius: BORDER_RADIUS,
    backgroundColor: 'rgba(255,255,255,0.8)',
    padding: PADDING,
  },

  cancelTextStyle: {
    textAlign: 'center',
    color: '#333',
    fontSize: FONT_SIZE,
  },

  optionStyle: {
    padding: PADDING,
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
  },
  optionIconStyle: {
    textAlign: 'center',
  },
  optionTextStyle: {
    textAlign: 'center',
    fontSize: FONT_SIZE,
    color: HIGHLIGHT_COLOR,
  },

  sectionStyle: {
    padding: PADDING * 2,
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
  },

  sectionTextStyle: {
    textAlign: 'center',
    fontSize: FONT_SIZE,
  },

  mapContainer: {
    borderRadius: BORDER_RADIUS,
    flexShrink: 1,
    marginTop: 8,
    marginBottom: 8,
    padding: PADDING,
    backgroundColor: 'rgba(255,255,255,0.8)',
  },

  map: {
    backgroundColor: '#f9f9f9',
    zIndex: -1,
    height: 200,
  },
});

export default styles;
