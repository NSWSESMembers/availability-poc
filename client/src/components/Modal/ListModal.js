import React, { Component } from 'react';

import PropTypes from 'prop-types';
import { View, Modal, TouchableOpacity, TouchableWithoutFeedback, Text, ScrollView } from 'react-native';

import styles from './styles';

let componentIndex = 0;

class ListModal extends Component {
  renderOption(option, isLastItem) {
    return (
      <TouchableOpacity key={option.key} onPress={() => this.props.onChange(option)}>
        <View style={[styles.optionStyle, isLastItem &&
            { borderBottomWidth: 0 }]}
        >
          <Text style={styles.optionTextStyle}>{option.label}</Text>
        </View>
      </TouchableOpacity>);
  }

  renderOptionList() {
    const options = this.props.data.map(
      (item, index) => this.renderOption(
        item, index === this.props.data.length - 1,
      ),
    );

    componentIndex += 1;

    return (
      <TouchableWithoutFeedback
        key={`modalSelector${componentIndex}`}
        onPress={() => this.props.closeModal}
      >
        <View style={styles.overlayStyle}>
          <View style={styles.headerContainer}>
            <TouchableOpacity>
              <View style={styles.headerStyle}>
                <Text style={styles.headerTextStyle}>{this.props.title}</Text>
              </View>
            </TouchableOpacity>
          </View>
          <View style={styles.optionContainer}>
            <ScrollView>
              <View style={{ paddingHorizontal: 10 }}>
                {options}
              </View>
            </ScrollView>
          </View>
          <View style={styles.cancelContainer}>
            <TouchableOpacity onPress={this.props.backModal}>
              <View style={styles.cancelStyle}>
                <Text style={styles.cancelTextStyle}>Back</Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
      </TouchableWithoutFeedback>);
  }


  render() {
    return (
      <View>
        <Modal
          transparent
          visible={this.props.visible}
          onRequestClose={this.props.closeModal}
          animationType="fade"
        >
          {this.renderOptionList()}
        </Modal>
      </View>
    );
  }
}
ListModal.propTypes = {
  title: PropTypes.string.isRequired,
  data: PropTypes.array, // eslint-disable-line react/forbid-prop-types
  closeModal: PropTypes.func.isRequired,
  backModal: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
};

export default ListModal;
