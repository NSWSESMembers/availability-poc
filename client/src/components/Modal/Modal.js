import React from 'react';

import PropTypes from 'prop-types';
import { View, Button, Modal } from 'react-native';

import styles from './styles';

const BasicModal = ({ visible, closeModal, children }) => (
  <Modal visible={visible} animationType="slide" onRequestClose={closeModal}>
    <View style={styles.container}>
      {children}
      <Button onPress={closeModal} title="Cancel" />
    </View>
  </Modal>
);

BasicModal.propTypes = {
  children: PropTypes.node,
  closeModal: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
};

export default BasicModal;
