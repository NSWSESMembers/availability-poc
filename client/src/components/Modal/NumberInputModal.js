import React from 'react';

import PropTypes from 'prop-types';
import { View, Modal, TouchableOpacity, TouchableWithoutFeedback, Text, TextInput } from 'react-native';

import styles from './styles';

const NumberInputModal = ({
  title,
  placeHolder,
  closeModal,
  backModal,
  onChangeText,
  visible,
  onSave,
}) => (
  <View>
    <Modal
      transparent
      visible={visible}
      onRequestClose={closeModal}
      animationType="fade"
    >
      <TouchableWithoutFeedback
        onPress={() => closeModal}
      >
        <View style={styles.overlayStyle}>
          <View style={styles.headerContainer}>
            <TouchableOpacity>
              <View style={styles.headerStyle}>
                <Text style={styles.headerTextStyle}>{title}</Text>
              </View>
            </TouchableOpacity>
          </View>
          <View style={styles.optionContainer}>
            <View>
              <View style={{ paddingHorizontal: 10 }}>
                <TextInput
                  style={styles.etaInput}
                  keyboardType="numeric"
                  placeholder={placeHolder}
                  maxLength={10}
                  onChangeText={onChangeText}
                />
              </View>
            </View>
          </View>
          <View style={styles.nextContainer}>
            <TouchableOpacity onPress={() => onSave}>
              <View style={styles.nextStyle}>
                <Text style={styles.nextTextStyle}>Next</Text>
              </View>
            </TouchableOpacity>
          </View>
          <View style={styles.cancelContainer}>
            <TouchableOpacity onPress={() => backModal}>
              <View style={styles.cancelStyle}>
                <Text style={styles.cancelTextStyle}>Back</Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
      </TouchableWithoutFeedback>
    </Modal>
  </View>
);

NumberInputModal.propTypes = {
  title: PropTypes.string.isRequired,
  placeHolder: PropTypes.string.isRequired,
  closeModal: PropTypes.func.isRequired,
  onSave: PropTypes.func.isRequired,
  backModal: PropTypes.func.isRequired,
  onChangeText: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
};

export default NumberInputModal;
