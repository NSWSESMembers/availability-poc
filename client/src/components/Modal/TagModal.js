import React, { Component } from 'react';
import CheckBox from 'react-native-check-box';

import PropTypes from 'prop-types';
import { View, Modal, TouchableOpacity, TouchableWithoutFeedback, Text, ScrollView } from 'react-native';
import { SearchBar } from 'react-native-elements';

import { Center } from '../Container';

import styles from './styles';

class TagModal extends Component {
  isChecked = (option) => {
    const index = this.props.selectedTags.indexOf(option.id);
    return index !== -1;
  };

  renderOption(option, isLastItem) {
    return (
      <TouchableOpacity key={option.id}>
        <View style={[styles.optionStyle, isLastItem && { borderBottomWidth: 0 }]}>
          <CheckBox
            style={{ flex: 1, padding: 10 }}
            rightText={option.name}
            onClick={() => this.props.onSelect(option)}
            isChecked={this.isChecked(option)}
          />
        </View>
      </TouchableOpacity>
    );
  }

  renderOptionList() {
    const options = this.props.dataIn.map((item, index) =>
      this.renderOption(item, index === this.props.dataIn.length - 1));

    if (options.length) {
      return [options];
    }
    return (
      <Center>
        <Text>No matching tags found</Text>
      </Center>
    );
  }

  render() {
    return (
      <View>
        <Modal
          transparent
          visible={this.props.visible}
          onRequestClose={this.props.closeModal}
          animationType="fade"
        >
          <TouchableWithoutFeedback
            onPress={this.props.closeModal}
          >
            <View style={styles.overlayStyle}>
              <View style={styles.headerContainer}>
                <TouchableOpacity>
                  <View style={styles.headerStyle}>
                    <Text style={styles.headerTextStyle}>{this.props.headerText}</Text>
                  </View>
                </TouchableOpacity>
              </View>
              <View style={styles.searchContainer}>
                <SearchBar
                  lightTheme
                  round
                  showLoading={this.props.isLoading}
                  containerStyle={styles.searchbar}
                  onChangeText={this.props.onSearch}
                  onClearText={null}
                  placeholder="Search"
                />
              </View>
              <View
                style={{
                borderBottomColor: 'grey',
                borderBottomWidth: 1,
                alignItems: 'center',
              }}
              />
              <View style={styles.headerContainer} />
              <View style={styles.optionContainerTags}>
                <ScrollView keyboardShouldPersistTaps="handled">
                  <View style={{ paddingHorizontal: 10 }}>{this.renderOptionList()}</View>
                </ScrollView>
              </View>
              <View style={styles.cancelContainer}>
                <TouchableOpacity onPress={this.props.backModal}>
                  <View style={styles.cancelStyle}>
                    <Text style={styles.cancelTextStyle}>Save</Text>
                  </View>
                </TouchableOpacity>
              </View>
            </View>
          </TouchableWithoutFeedback>
        </Modal>
      </View>
    );
  }
}
TagModal.propTypes = {
  dataIn: PropTypes.array, // eslint-disable-line react/forbid-prop-types
  selectedTags: PropTypes.array, // eslint-disable-line react/forbid-prop-types
  headerText: PropTypes.string.isRequired,
  closeModal: PropTypes.func.isRequired,
  onSearch: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
  isLoading: PropTypes.bool,
  backModal: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
};

export default TagModal;
