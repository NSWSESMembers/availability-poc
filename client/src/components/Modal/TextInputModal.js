import React from 'react';

import PropTypes from 'prop-types';
import { View, Modal, TouchableOpacity, TouchableWithoutFeedback, Text, TextInput } from 'react-native';

import styles from './styles';

const TextInputModal = ({ closeModal, backModal, onChangeText, visible, onSave }) => (
  <View>
    <Modal
      transparent
      visible={visible}
      onRequestClose={closeModal}
      animationType="fade"
    >
      <TouchableWithoutFeedback
        onPress={() => closeModal}
      >
        <View style={styles.overlayStyle}>
          <View style={styles.headerContainer}>
            <TouchableOpacity>
              <View style={styles.headerStyle}>
                <Text style={styles.headerTextStyle}>Availability Comments</Text>
              </View>
            </TouchableOpacity>
          </View>
          <View style={styles.optionContainer}>
            <View>
              <View style={{ paddingHorizontal: 10 }}>
                <TextInput
                  style={styles.etaInput}
                  placeholder="Notes.."
                  onChangeText={onChangeText}
                />
              </View>
            </View>
          </View>
          <View style={styles.nextContainer}>
            <TouchableOpacity onPress={onSave}>
              <View style={styles.nextStyle}>
                <Text style={styles.nextTextStyle}>Next</Text>
              </View>
            </TouchableOpacity>
          </View>
          <View style={styles.cancelContainer}>
            <TouchableOpacity onPress={backModal}>
              <View style={styles.cancelStyle}>
                <Text style={styles.cancelTextStyle}>Back</Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
      </TouchableWithoutFeedback>
    </Modal>
  </View>
);

TextInputModal.propTypes = {
  closeModal: PropTypes.func.isRequired,
  onSave: PropTypes.func.isRequired,
  backModal: PropTypes.func.isRequired,
  onChangeText: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
};

export default TextInputModal;
