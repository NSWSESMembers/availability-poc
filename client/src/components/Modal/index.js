import Modal from './Modal';
import ListModal from './ListModal';
import NumberInputModal from './NumberInputModal';
import TextInputModal from './TextInputModal';
import styles from './styles';

export { Modal, ListModal, NumberInputModal, TextInputModal, styles };
