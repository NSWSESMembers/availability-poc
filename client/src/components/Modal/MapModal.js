import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, Modal, TouchableOpacity, Dimensions, TouchableWithoutFeedback, Text } from 'react-native';

import MapView, { Marker } from 'react-native-maps';
import { IconMarker } from '../../components/MapMarker/';
import MapDelta from '../../selectors/MapDelta';

import styles from './styles';

const screen = Dimensions.get('window');

const ASPECT_RATIO = screen.width / screen.height;
const LATITUDE_DELTA = 0.0922;
const LONGITUDE_DELTA = LATITUDE_DELTA * ASPECT_RATIO;

class MapModal extends Component {
  zoomMap = () => {
    if (this.props.initialRegion) {
      this.map.animateToRegion(
        MapDelta(
          this.props.initialRegion.locationLatitude,
          this.props.initialRegion.locationLongitude,
          2000, // 2000m height
        ),
      );
    }
  }

  render() {
    return (
      <View>
        <Modal
          transparent
          visible={this.props.visible}
          onRequestClose={this.props.closeModal}
          animationType="fade"
        >
          <TouchableWithoutFeedback onPress={this.props.closeModal}>
            <View style={styles.overlayStyle}>
              <View style={styles.headerContainer}>
                <TouchableOpacity>
                  <View style={styles.headerStyle}>
                    <Text style={styles.headerTextStyle}>{this.props.title}</Text>
                  </View>
                </TouchableOpacity>
              </View>
              <View style={styles.mapContainer}>
                <View>
                  <View>
                    {/* TouchableWithoutFeedback prevents touch propagation */}
                    <TouchableWithoutFeedback>
                      <MapView
                        onMapReady={this.zoomMap}
                        ref={(ref) => {
                          this.map = ref;
                        }}
                        initialRegion={{
                          latitude: this.props.initialRegion.locationLatitude || 0,
                          longitude: this.props.initialRegion.locationLongitude || 0,
                          latitudeDelta: LATITUDE_DELTA,
                          longitudeDelta: LONGITUDE_DELTA,
                       }}
                        // TODO Dynamic height
                        style={{ height: Dimensions.get('window').height - 250 }}
                      >
                        {this.props.markers.map(marker => (
                          <Marker
                            key={`marker${marker.id}`}
                            coordinate={marker}
                            style={{ zIndex: 1, opacity: 0.9 }}
                          >
                            <IconMarker
                              name={marker.icon}
                            />
                          </Marker>
                        ))}
                      </MapView>
                    </TouchableWithoutFeedback>
                  </View>
                </View>
              </View>
              <View style={styles.cancelContainer}>
                <TouchableOpacity onPress={this.props.backModal}>
                  <View style={styles.cancelStyle}>
                    <Text style={styles.cancelTextStyle}>Back</Text>
                  </View>
                </TouchableOpacity>
              </View>
            </View>
          </TouchableWithoutFeedback>
        </Modal>
      </View>
    );
  }
}


MapModal.propTypes = {
  title: PropTypes.string.isRequired,
  markers: PropTypes.array, // eslint-disable-line react/forbid-prop-types
  initialRegion: PropTypes.shape({
    locationLatitude: PropTypes.number,
    locationLongitude: PropTypes.number,
  }),
  closeModal: PropTypes.func.isRequired,
  backModal: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
};

export default MapModal;
