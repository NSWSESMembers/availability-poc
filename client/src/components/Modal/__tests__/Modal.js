import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import Modal from '../Modal';

const noop = () => {};

const child = <Text>child</Text>;

test('renders correctly', () => {
  const tree1 = renderer
    .create(
      <Modal closeModal={noop} visible>
        {child}
      </Modal>,
    )
    .toJSON();
  expect(tree1).toMatchSnapshot();
});
