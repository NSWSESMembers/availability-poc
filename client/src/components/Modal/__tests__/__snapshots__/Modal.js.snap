// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  visible={true}
>
  <View
    style={undefined}
  >
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        child
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Close modal
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
